pkg/cvo/egress.go

```diff
@@ -21,14 +21,12 @@ func (optr *Operator) getHTTPSProxyURL() (*url.URL, error) {
 		return nil, err
 	}
 
-	if &proxy.Spec != nil {
```

Comment:
`proxy` is referenced and compared to nil. Since proxy is a value type this check will never be false.

Reply:
Looks good, but I'm not entirely sure if these are equivalent.

/cc @jottofar @wking

Reply:
I didn't change the check. I removed it and what you see now is the inner-check. The outer-check was superfluous for the reason mentioned above.